azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -289,9 +290,14 @@ def _update(self, force_update):
         # case, to ensure it fetches the new certificate.
         #
         if self._extensions_goal_state.source == GoalStateSource.FastTrack:
-            self._check_certificates()
+            certs_uri = findtext(xml_doc, "Certificates")
```

Comment:
Minor: let's add certs_uri as a data member of self and use that instead of picking it up  from the xml_doc we just fetched. With the current code, this makes no difference functionally, but let's do that in case the code changes. That would also emphasize that we are redownloading the same certs as we did before. 

Also, we should re-download only if self._goal_state_properties & GoalStateProperties.Certificates != 0